Cargo.toml

```diff
@@ -10,13 +10,14 @@ repository = "https://github.com/tafia/quick-xml"
 keywords = ["xml", "serde", "parser", "writer", "html"]
 categories = ["asynchronous", "encoding", "parsing", "parser-implementations"]
 license = "MIT"
+rust = "1.54"
 
 [dependencies]
 document-features = { version = "0.2", optional = true }
 encoding_rs = { version = "0.8", optional = true }
-serde = { version = "1.0", optional = true }
-tokio = { version = "1.21", optional = true, default-features = false, features = ["io-util"] }
-memchr = "2.5"
+serde = { version = "^1.0.100", optional = true }
```

Comment:
As far as I know, [caret requirements](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements) are the default and I would suggest omitting the carets here for consistency with the other dependencies, otherwise it looks like there is a difference when there actually is not.

Reply:
Agree. Also, why `1.0.100`? Something required for us in that version was introduced? If so, then maybe `>=1.0.100`?

Reply:
> Agree. Also, why `1.0.100`? Something required for us in that version was introduced? If so, then maybe `>=1.0.100`?

The default and caret requirements already mean `>=1.0.100, <2`.

On why `1.0.100`, serde is known for introducing new functionality in patch releases which suggests to specify a minimum patch release one has actually tested (or ideally even figuring out the true minimum patch release), c.f. https://users.rust-lang.org/t/psa-please-specify-precise-dependency-versions-in-cargo-toml/71277

Reply:
>As far as I know, [caret requirements](https://doc.rust-lang.org/cargo/reference/specifying-dependencies.html#caret-requirements) are the default and I would suggest omitting the carets here for consistency with the other dependencies, otherwise it looks like there is a difference when there actually is not.

The "crates" extension for VSCode was telling me that the unadorned version would prefer the oldest, whereas `^$version` selects the newest.  However, it is entirely possible that this is incorrect or at least misleading.  If it's documented that they are equivalent I have no reason to doubt that.

>Agree. Also, why 1.0.100? Something required for us in that version was introduced? If so, then maybe >=1.0.100?

I tried with serde 1.0 but it seemed to be using some nightly features that no longer exist.  There's nothing special about .100 other than that it is reasonably old and worked, and I didn't feel like bisecting 144 different versions at 1 AM.  I can do that now if we feel strongly about it.

Reply:
>  I can do that now if we feel strongly about it.

No, it's fine, but I would prefer if you add this consideration in comment in `Cargo.toml` for future readers

Reply:
>There's nothing special about .100 other than that it is reasonably old and worked, and I didn't feel like bisecting 144 different versions at 1 AM. I can do that now if we feel strongly about it.

It turns out that no, out of all 144 versions, .100 **is** actually the minimum that works.  So my random selection was the actual right answer.  I should go buy a lottery ticket..